tests/Type/WebMozartAssert/data/bug-17.php

```diff
@@ -0,0 +1,11 @@
+<?php declare(strict_types=1);
+
+namespace Bug17;
+
+use DateTimeInterface;
+use Webmozart\Assert\Assert;
+
+(function () {
+	Assert::implementsInterface(\DateTime::class, DateTimeInterface::class);
+	Assert::implementsInterface(\DateTimeZone::class, DateTimeInterface::class);
```

Comment:
this should/could always evaluate to false? See https://phpstan.org/r/0f9382f8-27c1-4f60-84ce-434da33d0e32

Reply:
oh, interesting, this is reported as false with lowest deps, but with highest it's not. a regression in phpstan?

Reply:
the behaviour changed with https://github.com/phpstan/phpstan/releases/tag/1.8.1

Reply:
ah, looks like
> `GenericClassStringType::isSuperTypeOf()` - do not convert maybe into no (https://github.com/phpstan/phpstan-src/commit/6fba5429b9784801a89cdc6cfa3d6f8cfd1d82e8), https://github.com/phpstan/phpstan/issues/6697

I guess I have to assume this is a fix then

Reply:
I'm still not sure, maybe it is a regression after all..

Reply:
nah, this is the "subclass could still implement it" problem again, got it.. it should be good then